### PR TITLE
[Build] Add new dependencies to Maestro

### PR DESCRIPTION
**Pull request by rmarinho**
### Description of Change ###

Adding new versions

### Additions made ###

* Adds 

### PR Checklist ###

- [ ] Targets the correct branch 
- [ ] Tests are passing (or failures are unrelated)
- [ ] Targets a single property for a single control (or intertwined few properties)
- [ ] Adds the property to the appropriate interface
- [ ] Avoids any changes not essential to the handler property
- [ ] Adds the mapping to the PropertyMapper in the handler
- [ ] Adds the mapping method to the Android, iOS, and Standard aspects of the handler
- [ ] Implements the actual property updates (usually in extension methods in the Platform section of Core)
- [ ] Tags ported renderer methods with [PortHandler]
- [ ] Adds an example of the property to the sample project (MainPage)
- [ ] Adds the property to the stub class
- [ ] Implements basic property tests in DeviceTests

#### Does this PR touch anything that might affect accessibility?
- [ ] Does this PR introduce a new control? (If yes, add an example using SemanticProperties to the SemanticsPage)
- [ ] APIs that modify focusability?
- [ ] APIs that modify any text property on a control?
- [ ] Does this PR modify view nesting or view arrangement in anyway?
- [ ] Is there the smallest possibility that your PR will change accessibility? 
- [ ] I'm not sure, please help me

If any of the above checkboxes apply to your PR, then the PR will need to provide testing to demonstrate that accessibility still works.

### CODE REVIEW

File: eng/Version.Details.xml [hunk 1]
```diff
@@ -24,5 +24,94 @@
       <Uri>https://github.com/xamarin/xamarin-macios</Uri>
       <Sha>e5d12ecc90cad2b588e454ebec36556658981b28</Sha>
     </Dependency>
+    <Dependency Name="Microsoft.AspNetCore.App.Runtime.win-x64" Version="6.0.0-preview.5.21301.17" CoherentParentDependency="Microsoft.Dotnet.Sdk.Internal">
+      <Uri>https://github.com/dotnet/aspnetcore</Uri>
+      <Sha>5a007deadbb2636caff4656333cafc818b6b2ef6</Sha>
+    </Dependency>
+    <Dependency Name="Microsoft.Extensions.Hosting.Abstractions" Version="6.0.0-preview.5.21301.5" CoherentParentDependency="Microsoft.AspNetCore.App.Runtime.win-x64">
+      <Uri>https://github.com/dotnet/runtime</Uri>
```
Comment (Eilon) [inline eng/Version.Details.xml:32]: To anyone else looking: The URI's and SHA's here are arbitrary and don't matter. Maestro will update them when it next runs.

**Review by mmitche — APPROVED** [review]

**Comment by rmarinho** [comment]
The oens tha tneed to go to the installer 

```
 Unable to update Microsoft.Extensions.FileProviders.Embedded to have coherency with Microsoft.Dotnet.Sdk.Internal: https://github.com/dotnet/installer @ d6380bcae783600a01a9c8a6a151bb6a66c777b0 does not contain dependency Microsoft.Extensions.FileProviders.Embedded
    - Add the dependency to https://github.com/dotnet/installer.
    - Pin the dependenency.
    - Remove the CoherentParentDependency attribute.
  Unable to update Microsoft.AspNetCore.Authorization to have coherency with Microsoft.Dotnet.Sdk.Internal: https://github.com/dotnet/installer @ d6380bcae783600a01a9c8a6a151bb6a66c777b0 does not contain dependency Microsoft.AspNetCore.Authorization
    - Add the dependency to https://github.com/dotnet/installer.
    - Pin the dependenency.
    - Remove the CoherentParentDependency attribute.
  Unable to update Microsoft.AspNetCore.Components.WebView to have coherency with Microsoft.Dotnet.Sdk.Internal: https://github.com/dotnet/installer @ d6380bcae783600a01a9c8a6a151bb6a66c777b0 does not contain dependency Microsoft.AspNetCore.Components.WebView
    - Add the dependency to https://github.com/dotnet/installer.
    - Pin the dependenency.
    - Remove the CoherentParentDependency attribute.
  Unable to update Microsoft.AspNetCore.Components.Web to have coherency with Microsoft.Dotnet.Sdk.Internal: https://github.com/dotnet/installer @ d6380bcae783600a01a9c8a6a151bb6a66c777b0 does not contain dependency Microsoft.AspNetCore.Components.Web
    - Add the dependency to https://github.com/dotnet/installer.
    - Pin the dependenency.
    - Remove the CoherentParentDependency attribute.
  Unable to update Microsoft.JSInterop to have coherency with Microsoft.Dotnet.Sdk.Internal: https://github.com/dotnet/installer @ d6380bcae783600a01a9c8a6a151bb6a66c777b0 does not contain dependency Microsoft.JSInterop
    - Add the dependency to https://github.com/dotnet/installer.
    - Pin the dependenency.
    - Remove the CoherentParentDependency attribute.
  Unable to update Microsoft.Web.WebView to have coherency with Microsoft.Dotnet.Sdk.Internal: https://github.com/dotnet/installer @ d6380bcae783600a01a9c8a6a151bb6a66c777b0 does not contain dependency Microsoft.Web.WebView
    - Add the dependency to https://github.com/dotnet/installer.
    - Pin the dependenency.
    - Remove the CoherentParentDependency attribute.
```

**Review by Eilon — APPROVED** [review]

**Comment by mmitche** [comment]
@Eilon @rmarinho Microsoft.Web.WebView isn't a dependency that Maestro knows anything about. Where it supposed to come from? The project appears to actually reference Microsoft.Web.WebView2, but we don't know about that package either.

**Comment by Eilon** [comment]
`Microsoft.Web.WebView` is an external dependency from the Edge browser team. It comes from NuGet.org (mirrored in the internal feeds).

File: eng/Version.Details.xml [hunk 2]
```diff
@@ -24,5 +24,65 @@
...
+    </Dependency>
+    <Dependency Name="Microsoft.AspNetCore.Components.Web" Version="6.0.0-preview.5.21301.5" CoherentParentDependency="Microsoft.Dotnet.Sdk.Internal">
+      <Uri>https://github.com/dotnet/runtime</Uri>
+      <Sha>ec3e0b276b159ab24946f17f2b89d00862488a6a</Sha>
+    </Dependency>
+    <Dependency Name="Microsoft.JSInterop" Version="6.0.0-preview.5.21301.5" CoherentParentDependency="Microsoft.Dotnet.Sdk.Internal">
+      <Uri>https://github.com/dotnet/runtime</Uri>
+      <Sha>ec3e0b276b159ab24946f17f2b89d00862488a6a</Sha>
+    </Dependency>
+      <Dependency Name="Microsoft.Web.WebView" Version="6.0.0-preview.5.21301.5" CoherentParentDependency="Microsoft.Dotnet.Sdk.Internal">
```
Comment (Eilon) [inline eng/Version.Details.xml:79]: Sorry I missed this earlier. This entry shouldn't be in here because it's not part of .NET 6.

**Review by Eilon — CHANGES_REQUESTED** [review]
Need to remove WebView2 from the list.

File: eng/Version.Details.xml (hunk 2 above)
Comment (rmarinho) [reply eng/Version.Details.xml]: Fixed

**Review by Eilon — APPROVED** [review]

**Review by jsuarezruiz — APPROVED** [review]

File: eng/Versions.props [hunk 3]
```diff
@@ -2,12 +2,24 @@
...
   <PropertyGroup>
     <MicrosoftDotnetSdkInternalPackageVersion>6.0.100-preview.6.21313.2</MicrosoftDotnetSdkInternalPackageVersion>
-    <MicrosoftNETCoreAppRefPackageVersion>6.0.0-preview.6.21306.1</MicrosoftNETCoreAppRefPackageVersion>
+    <MicrosoftNETCoreAppRefPackageVersion>6.0.0-preview.6.21307.1</MicrosoftNETCoreAppRefPackageVersion>
     <MicrosoftAndroidSdkWindowsPackageVersion>30.0.100-ci.main.40</MicrosoftAndroidSdkWindowsPackageVersion>
     <MicrosoftMacCatalystSdkPackageVersion>15.0.100-ci.main.25</MicrosoftMacCatalystSdkPackageVersion>
     <MicrosoftiOSSdkPackageVersion>15.0.100-ci.main.25</MicrosoftiOSSdkPackageVersion>
     <MicrosofttvOSSdkPackageVersion>15.0.100-ci.main.25</MicrosofttvOSSdkPackageVersion>
     <MicrosoftmacOSSdkPackageVersion>12.0.100-ci.main.25</MicrosoftmacOSSdkPackageVersion>
+    <MicrosoftExtensionsHostingAbstractions>12.0.100-ci.main.25</MicrosoftExtensionsHostingAbstractions>
```
Comment (Eilon) [inline eng/Versions.props:11]: I assume these are also just "random" values that will be overwritten by Maestro when it comes around.

Comment (mmitche) [reply eng/Versions.props:11]: You need to suffix these with either "Version" or "PackageVersion".

Also it won't automatically fix what doesn't need updating in the version.details.xml file, so keep that in mind.

`darc verify` can tell you what might be wrong.

Comment (rmarinho) [reply eng/Versions.props]: Yes thanks, it's fixed and versions updated now

File: eng/Version.Details.xml (hunk 1 above)
Comment (rmarinho) [reply eng/Version.Details.xml:36]: fixed

**Comment by Eilon** [comment]
> You need to suffix these with either "Version" or "PackageVersion".

Do we need to pick which one is right, or is only one or the other correct in a given package name? Because if we get to pick, I would think these would all be PackageVersion because we need to know the NuGet package version (as opposed to, say, an assembly version?).

**Review by Eilon — APPROVED** [review]
I re-sign-off!

File: eng/Microsoft.Extensions.targets [hunk 4]
```diff
@@ -3,24 +3,41 @@
...
-    <_MicrosoftFileProvidersVersion>5.0.6</_MicrosoftFileProvidersVersion>
     <_MicrosoftSystemCodeDomVersion>4.7.0</_MicrosoftSystemCodeDomVersion>
+    <_MicrosoftExtensionsHostingAbstractionsPackageVersion>$(_MicrosoftHostingVersion)</_MicrosoftExtensionsHostingAbstractionsPackageVersion>
+    <_MicrosoftExtensionsDependencyInjectionPackageVersion>5.0.1</_MicrosoftExtensionsDependencyInjectionPackageVersion>
+    <_MicrosoftExtensionsConfigurationPackageVersion>$(_MicrosoftHostingVersion)</_MicrosoftExtensionsConfigurationPackageVersion>
+    <_MicrosoftExtensionsLoggingAbstractionsPackageVersion>$(_MicrosoftHostingVersion)</_MicrosoftExtensionsLoggingAbstractionsPackageVersion>
+    <_MicrosoftExtensionsLoggingPackageVersion>$(_MicrosoftHostingVersion)</_MicrosoftExtensionsLoggingPackageVersion>
+    <_MicrosoftExtensionsLoggingConsolePackageVersion>$(_MicrosoftHostingVersion)</_MicrosoftExtensionsLoggingConsolePackageVersion>
+    <_MicrosoftExtensionsLoggingDebugPackageVersion>$(MicrosoftExten_MicrosoftHostingVersionsionsLoggingDebugPackageVersion)</_MicrosoftExtensionsLoggingDebugPackageVersion>
```
Comment (Eilon) [inline eng/Microsoft.Extensions.targets:25]: Typo in the value here.

**Comment by Eilon** [comment]
I quickly tried WinUI+Blazor and it works. I had to manually install a build of [.NET 6 Preview 6](https://aka.ms/dotnet/6.0/daily/dotnet-sdk-win-x64.exe) (note: this link is "latest", so if you're in the future, you'll get a different build from me) because the WinUI solution doesn't use an in-place .NET runtime.

image

**Comment by rmarinho** [comment]
humm but builds are failing

File: eng/Microsoft.Extensions.targets [hunk 5]
```diff
@@ -3,24 +3,47 @@
...
+    <_MicrosoftExtensionsLoggingConsolePackageVersion>$(MicrosoftExtensionsLoggingConsolePackageVersion)</_MicrosoftExtensionsLoggingConsolePackageVersion>
+    <_MicrosoftExtensionsLoggingDebugPackageVersion>$(MicrosoftExtensionsLoggingDebugPackageVersion)</_MicrosoftExtensionsLoggingDebugPackageVersion>
+    <_MicrosoftExtensionsFileProvidersAbstractionsPackageVersion>$(MicrosoftExtensionsFileProvidersAbstractionsPackageVersion)</_MicrosoftExtensionsFileProvidersAbstractionsPackageVersion>
+    <_MicrosoftExtensionsFileProvidersEmbeddedPackageVersion>$(MicrosoftExtensionsFileProvidersEmbeddedPackageVersion)</_MicrosoftExtensionsFileProvidersEmbeddedPackageVersion>
   </PropertyGroup>
   <PropertyGroup Condition=" '$(_UseNet6Packages)' != 'true' ">
     <_MicrosoftHostingVersion>5.0.0</_MicrosoftHostingVersion>
-    <_MicrosoftDependencyInjectionVersion>5.0.1</_MicrosoftDependencyInjectionVersion>
-    <_MicrosoftFileProvidersVersion>5.0.6</_MicrosoftFileProvidersVersion>
     <_MicrosoftSystemCodeDomVersion>4.7.0</_MicrosoftSystemCodeDomVersion>
```
Comment (jonathanpeppers) [inline eng/Microsoft.Extensions.targets:21]: Does this one need to be remove and we use `$(SystemCodeDomPackageVersion)` instead?

File: eng/Microsoft.Extensions.targets [hunk 6]
```diff
@@ -3,24 +3,47 @@
...
+    <_MicrosoftExtensionsConfigurationPackageVersion>$(MicrosoftExtensionsConfigurationPackageVersion)</_MicrosoftExtensionsConfigurationPackageVersion>
+    <_MicrosoftExtensionsLoggingAbstractionsPackageVersion>$(MicrosoftExtensionsLoggingAbstractionsPackageVersion)</_MicrosoftExtensionsLoggingAbstractionsPackageVersion>
+    <_MicrosoftExtensionsLoggingPackageVersion>$(MicrosoftExtensionsLoggingPackageVersion)</_MicrosoftExtensionsLoggingPackageVersion>
+    <_MicrosoftExtensionsLoggingConsolePackageVersion>$(MicrosoftExtensionsLoggingConsolePackageVersion)</_MicrosoftExtensionsLoggingConsolePackageVersion>
+    <_MicrosoftExtensionsLoggingDebugPackageVersion>$(MicrosoftExtensionsLoggingDebugPackageVersion)</_MicrosoftExtensionsLoggingDebugPackageVersion>
+    <_MicrosoftExtensionsFileProvidersAbstractionsPackageVersion>$(MicrosoftExtensionsFileProvidersAbstractionsPackageVersion)</_MicrosoftExtensionsFileProvidersAbstractionsPackageVersion>
+    <_MicrosoftExtensionsFileProvidersEmbeddedPackageVersion>$(MicrosoftExtensionsFileProvidersEmbeddedPackageVersion)</_MicrosoftExtensionsFileProvidersEmbeddedPackageVersion>
```
Comment (jonathanpeppers) [inline eng/Microsoft.Extensions.targets:17]: Can we just use the properties like `$(MicrosoftExtensionsFileProvidersEmbeddedPackageVersion)` directly and not make new properties?

Comment (Eilon) [reply eng/Microsoft.Extensions.targets:17]: I was wondering this too. For the non-net6 builds we need to use different versions for some packages, so we need _some_ strategy to override them. We could do that by just overriding the versions if not-net6 and using the same variables.

Comment (rmarinho) [reply eng/Microsoft.Extensions.targets:17]: what if we stop using maestro and don't a version.props? that's why i created variables for each

Comment (rmarinho) [reply eng/Microsoft.Extensions.targets:17]: if i $(MicrosoftExtensionsFileProvidersEmbeddedPackageVersion) for non net6 isn't also the order important? i m not a expert of msbuild so not sure tbh

**Review by jonathanpeppers — CHANGES_REQUESTED** [review]
I don't think we should merge until this is solved:
```
 error XAGCANSF7004: System.ArgumentException: An entry with the same key already exists.
 ```
Merging this and we can work on another Maestro bump: https://github.com/xamarin/xamarin-android/pull/6032

**Comment by jonathanpeppers** [comment]
Going to `/azp run` this one, we have this merged now: https://github.com/dotnet/maui/pull/1363

**Comment by jonathanpeppers** [comment]
It looks like the issue is:
```
Skipping duplicate assembly: obj\Release\net6.0-android\android-x86\linked\System.Runtime.CompilerServices.Unsafe.dll
```

Somehow this package is now being used, and it wasn't before:
```
~\.nuget\packages\system.runtime.compilerservices.unsafe\6.0.0-preview.6.21307.1\lib\netcoreapp3.1\System.Runtime.CompilerServices.Unsafe.dll
```

I'll need to get a fix for this in xamarin-android. Unless there is a way to prevent this NuGet package from being used?

**Comment by rmarinho** [comment]
Why wouldn't we see this problem before? We updated a version of a package that now pulls this ?

**Comment by jonathanpeppers** [comment]
Yes, I believe so.

**Comment by jonathanpeppers** [comment]
Now that we had the logging to see what was wrong, I was able to reproduce the real problem in a test and fix it:

https://github.com/xamarin/xamarin-android/pull/6043

**Pull request merged**